node-notes-cli/app.js

```diff
@@ -0,0 +1,42 @@
+// get data obj
+const data = require('./data');
+const fs = require('fs');
+
+// show all notes when user's input === read:
+if (process.argv[2] === 'read') {
+  for (const key in data.notes) {
+    console.log(`${key}: ${data.notes[key]}`);
+  }
+}
+
+// add a note
+if (process.argv[2] === 'create') {
+  const newNote = process.argv[3];
+
+  data.notes[data.nextId] = newNote;
+  data.nextId++;
+  fs.writeFile('data.json', JSON.stringify(data, null, 2), err => {
+    if (err) throw err;
+  });
+}
+
+// delete a file
+// find a note by propery key of obj
+if (process.argv[2] === 'delete') {
+  const idToDelete = process.argv[3];
+  delete data.notes[idToDelete];
+  data.nextId--;
+  fs.writeFile('data.json', JSON.stringify(data, null, 2), err => {
```

Comment:
We should not reset the id ever. If a note is deleted, its id should be skipped forever.

Reply:
<img width="921" alt="Screen Shot 2022-08-18 at 12 41 13 PM" src="https://user-images.githubusercontent.com/106217355/185480431-b050d15c-9752-49ed-ba14-afd862f56279.png">